CMakeLists.txt

```diff
@@ -373,30 +373,6 @@ if (NOT _ONEDPL_PSTL_OFFLOAD STREQUAL off)
         message(STATUS "Adding ${FSYCL_PSTL_OFFLOAD_OPTION} compiler option")
         target_compile_options(oneDPL INTERFACE ${FSYCL_PSTL_OFFLOAD_OPTION})
         target_link_libraries(oneDPL INTERFACE ${FSYCL_PSTL_OFFLOAD_OPTION})
-    else()
-        # TODO: remove this "manual" -fsycl-pstl-offload suboption substitution and replace with FATAL_ERROR
-        # message(FATAL_ERROR "${CMAKE_CXX_COMPILER} doesn't support ${FSYCL_PSTL_OFFLOAD_OPTION}")
```

Comment:
Since PSTL Offload would be removed from oneDPL totally, adding this FATAL_ERROR makes no sense for me. 
The compiler flags logic would be handled on the PSTL Offload side now.

Reply:
So, what I think you are saying then is that we wont be testing PSTL_OFFLOAD initiated from within oneDPL cmake anymore? If so, should we remove the rest of the logic surrounding PSTL_OFFLOAD in this cmake file?

Alternatively, if we still want to have the option `-D_ONEDPL_PSTL_OFFLOAD`, (which I think we might), then I think we shouldn't allow it to be "silently skipped", for compilers which don't support it.  We have similar logic and fatal error if `omp` backend is specified, but `OpenMP_CXX_FOUND` is false after we `find_package(OpenMP)`.

Reply:
Ideally, we don't want to test PSTL Offload from oneDPL. 
But currently some testing logic was kept inside oneDPL test CMakes to allow us to build PSTL Offload testing infrastructure on top of it. We are still using a log of oneDPL test files, but not directly from oneDPL testing CMakes.
I would now consider `-D_ONEDPL_PSTL_OFFLOAD` as an internal flag required for interoperability with PSTL Offload infrastructure. 
Following that logic, the compiler support checks should be performed on the PSTL Offload side.

Reply:
OK, fair enough. We can leave it as is.